src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -19850,42 +19865,42 @@ void func_8085AACC(PlayState* play, Player* this, UNK_TYPE arg2) {
         }
     }
 
-    this->csMode = PLAYER_CSMODE_0;
+    this->csMode = PLAYER_CSMODE_NONE;
     this->unk_AA5 = PLAYER_UNKAA5_0;
 }
 
-void Player_Cutscene_SetPosAndYawToStart(Player* this, CsCmdActorCue* playerAction) {
-    this->actor.world.pos.x = playerAction->startPos.x;
-    this->actor.world.pos.y = playerAction->startPos.y;
-    this->actor.world.pos.z = playerAction->startPos.z;
+void Player_Cutscene_SetPosAndYawToStart(Player* this, CsCmdActorCue* cue) {
+    this->actor.world.pos.x = cue->startPos.x;
+    this->actor.world.pos.y = cue->startPos.y;
+    this->actor.world.pos.z = cue->startPos.z;
 
-    this->currentYaw = this->actor.shape.rot.y = playerAction->rot.y;
+    this->currentYaw = this->actor.shape.rot.y = cue->rot.y;
 }
 
-void Player_Cutscene_8085ABA8(Player* this, CsCmdActorCue* playerAction) {
-    f32 xDiff = playerAction->startPos.x - (s32)this->actor.world.pos.x;
-    f32 yDiff = playerAction->startPos.y - (s32)this->actor.world.pos.y;
-    f32 zDiff = playerAction->startPos.z - (s32)this->actor.world.pos.z;
+void Player_Cutscene_8085ABA8(Player* this, CsCmdActorCue* cue) {
+    f32 xDiff = cue->startPos.x - (s32)this->actor.world.pos.x;
+    f32 yDiff = cue->startPos.y - (s32)this->actor.world.pos.y;
+    f32 zDiff = cue->startPos.z - (s32)this->actor.world.pos.z;
     f32 dist;
     s16 temp_v0;
 
-    temp_v0 = (s16)playerAction->rot.y - this->actor.shape.rot.y;
+    temp_v0 = (s16)cue->rot.y - this->actor.shape.rot.y;
     dist = sqrtf(SQ(xDiff) + SQ(yDiff) + SQ(zDiff));
     if (this->linearVelocity == 0.0f) {
         if ((dist > 50.0f) || (ABS_ALT(temp_v0) > 0x4000)) {
-            Player_Cutscene_SetPosAndYawToStart(this, playerAction);
+            Player_Cutscene_SetPosAndYawToStart(this, cue);
         }
     }
 
     this->skelAnime.moveFlags = 0;
     func_8082E784(this);
 }
 
-void func_8085AC9C(PlayState* play, Player* this, CsCmdActorCue* cue, struct_8085DA94* arg3) {
-    if (arg3->type > 0) {
-        D_8085D990[arg3->type](play, this, arg3->anim);
-    } else if (arg3->type < 0) {
-        arg3->func(play, this, cue);
+void func_8085AC9C(PlayState* play, Player* this, CsCmdActorCue* cue, PlayerCsModeEntry* csEntry) {
+    if (csEntry->type >= PLAYER_CSTYPE_ANIM_1) {
```

Comment:
Actually wouldn't this make more sense with a `> NONE` check instead?

Reply:
I had it as that before, and perhaps that makes more sense. The thought process I had is that when an enum has different parts to it and it's checking a range, that sometimes it makes more sense to check it inclusive. Ex. for the enum:

```c
    ...
    /* 0x12 */ PAUSE_STATE_GAMEOVER_10,
    /* 0x13 */ PAUSE_STATE_OWLWARP_0,
    /* 0x14 */ PAUSE_STATE_OWLWARP_1,
    /* 0x15 */ PAUSE_STATE_OWLWARP_2,
    /* 0x16 */ PAUSE_STATE_OWLWARP_3,
    /* 0x17 */ PAUSE_STATE_OWLWARP_SELECT,
    /* 0x18 */ PAUSE_STATE_OWLWARP_CONFIRM,
    /* 0x19 */ PAUSE_STATE_OWLWARP_6,
    /* 0x1A */ PAUSE_STATE_UNPAUSE_SETUP,
    ...
```

The check `(state >= PAUSE_STATE_OWLWARP_0) && (state <= PAUSE_STATE_OWLWARP_6)` makes more senes than:
`(state > PAUSE_STATE_GAMEOVER_10) && (state < PAUSE_STATE_UNPAUSE_SETUP)`

So the idea was that by having `>= PLAYER_CSTYPE_ANIM_1`, it could tell you that this includes the ANIM cs types. But maybe since its only `>=` and not `>= && <=`, `> NONE` probably makes more sense

Reply:
I see. It is a bit to know which one to use when the `NONE` value is in the question.
Personally I tend to give priority to the `NONE` check but it is fine either way